cmd/snap-confine/mount-support-nvidia.c

```diff
@@ -145,10 +148,35 @@ static void sc_populate_libgl_with_hostfs_symlinks(const char *libgl_dir,
 	for (size_t i = 0; i < glob_res.gl_pathc; ++i) {
 		char symlink_name[512] = { 0 };
 		char symlink_target[512] = { 0 };
+		char prefix_dir[512] = { 0 };
 		const char *pathname = glob_res.gl_pathv[i];
 		char *pathname_copy
 		    SC_CLEANUP(sc_cleanup_string) = strdup(pathname);
+		char *dirname_copy
+		    SC_CLEANUP(sc_cleanup_string) = strdup(pathname);
```

Comment:
Did you mean to strdup pathname again? It looks like a bug

Reply:
Yes I totally mean it. POSIX `dirname()` modifies its input argument.

Reply:
Can you call those pathname_copy1 and pathname_copy2 and add a comment describing why we need them. A naive quick look (I agree this is right but just confusing) may mistake this for copy-paste error.

Reply:
Force pushed